src/databricks/labs/ucx/queries/views/latest_historical_per_workspace.sql

```diff
@@ -0,0 +1,17 @@
+WITH last_workflow_run AS (
+    SELECT
+      workspace_id,
+      MAX(STRUCT(finished_at, workflow_run_attempt, started_at, workflow_run_id)) AS max_struct
+    FROM $inventory.workflow_runs  -- $inventory is a hardcoded name for replacing target schema in a view definition
+    WHERE workflow_name = 'migration-progress-experimental'
+    GROUP BY workspace_id
```

Comment:
This will locate the latest workflow run for each workspace, but it won't return the _set of latest runs_ for each workspace.

Workflow runs will not only be recorded by the migration-progress job, but also `migrate-tables` (which updates some of the inventory). As things stand, when `migrate-tables` does a run that only updates the grants and migration-status inventory, this query will only return _that_ run-id and downstream queries will therefore ignore all the other record-types that weren't refreshed by that job.

Circling back, for each workspace it needs to return the latest run id for each job in the workspace.

Reply:
The where covers this right?     WHERE workflow_name = 'migration-progress-experimental'

Reply:
Ah, you're right… of-course that will be fine.

That said I thought we want to query the latest known data whereas here we will be ignoring some to an extent.

Reply:
yeah, for now we only update the UCX catalog tables in the migration progress. If we change that, we can update the query at that time.